tests/Tests.AzureAppConfiguration/TestHelper.cs

```diff
@@ -126,6 +128,36 @@ public static List<ConfigurationSetting> LoadJsonSettingsFromFile(string path)
             }
             return _kvCollection;
         }
+
+        public static bool ValidateLoggedError(Mock<ILogger> logger, string expectedMessage)
+        {
+            Func<object, Type, bool> state = (v, t) => v.ToString().StartsWith(expectedMessage);
+
+            logger.Verify(
+                x => x.Log(
+                    It.Is<LogLevel>(l => l == LogLevel.Warning),
+                    It.IsAny<EventId>(),
+                    It.Is<It.IsAnyType>((v, t) => state(v, t)),
+                    It.IsAny<Exception>(),
+                    It.Is<Func<It.IsAnyType, Exception, string>>((v, t) => true)));
+
+            return true;
+        }
+
+        public static bool ValidateLoggedSuccess(Mock<ILogger> logger, string expectedMessage)
+        {
+            Func<object, Type, bool> state = (v, t) => v.ToString().Contains(expectedMessage);
+
+            logger.Verify(
+                x => x.Log(
+                    It.Is<LogLevel>(l => l == LogLevel.Information || l == LogLevel.Debug),
```

Comment:
Can we split the debug and information checks? How about creating 3 helpers instead of 2
 - ValidateLoggedWarning
 - ValidateLoggedInformation 
 - ValidateLoggedDebug